core/src/main/scala/org/apache/spark/sql/SnappyDDLParser.scala

```diff
@@ -508,6 +511,14 @@ abstract class SnappyDDLParser(session: SparkSession)
     SET ~ (
         CURRENT.? ~ SCHEMA ~ '='.? ~ ws ~ identifier ~>
             ((schemaName: String) => SetSchema(schemaName)) |
+        DISABLE ~ PLAN ~ CACHING ~ (ALL ~ push(true)).? ~>
+            ((all: Any) => {
+              val allSessions = all match {
+                case None => false
+                case Some(_) => true
+              }
+              SetDisablePlanCaching(allSessions)
+            }) |
```

Comment:
Why a new parser rule? IMO "SET DISABLE PLAN CACHING" looks odd in either case. Why not just have a session property like snappydata.planCaching so "set snappydata.planCaching=true/false" will be fine. You will need to add the key to the list in SnappyConf.keyUpdateActions for which existing plan cache is disabled.

Reply:
@rishitesh and @swale answer to both questions from you too is same:

i.e. I wanted to provide something like DISABLE ALL.
Let's say there is a system and multiple sessions are there and similar type of queries are being run from different session. Actually most common use case will be like this where different clients are working and issuing similar queries. Now if we found out in a running session that there is something wrong with plan caching of the type of query they are using then they can just get hold of one session and disable it right across. If it is session wise then all the user has to issue this individually on their session.
If we want to maintain it just single session wise then I can remove ALL and adopt the simpler one. Please suggest.

Reply:
@kneeraj Don't quite understand why global setting is related to parser changes. I don't see SetDisablePlanCachingCommand doing anything different for "all" in this PR so it won't work in any case.

For global property you can have a separate property, then set it on SparkConf or System.setProperty in SnappyConf.keyUpdateActions (like already done for SchedulerPool), then the default can read the global setting when initializing SnappySession.disablePlanCaching:

```scala
private[sql] var disablePlanCaching: Boolean = sessionState.conf.getConfString(<sessionKey>,
    sparkContext.conf.get(<globalKey>, "false")).toBoolean
```

For the global case you will also need to traverse all open sessions and change which I guess you were asking. But I think it is reasonable that global setting only applies to all new sessions while all open sessions will have to change theirs explicitly. Otherwise there will be unnecessary overhead of maintaining/cleaning global list or else the disablePlanCaching will have to become a function which is again unnecessary overhead to cater to a very unlikely case.

Reply:
@swale yes that change is not there because I was reluctant to add global list and was just thinking how else I can do it and for now just go ahead with single session. Yes for the same reason I was asking about how to get the list of all active sessions a few days back.

Ok will change as suggested.